src/application-status-overlay/ApplicationStatusOverlay.jsx

```diff
@@ -0,0 +1,69 @@
+import { useRef, useContext, useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import uuidv4 from 'uuid/v4';
+import ApplicationStatusOverlayContext from './ApplicationStatusOverlayContext';
+
+const propTypes = {
+  /**
+   * An array of objects containing terra-button properties. A key attribute is required for each object.
+   * This array is used to render buttons in the bottom section.
+   * Example:`[{ text: 'Button 1', key: 1, size: 'medium', variant: 'action', onClick: onClickFunction}]`
+   */
+  // eslint-disable-next-line react/forbid-foreign-prop-types
+  buttonAttrs: PropTypes.arrayOf(PropTypes.shape(Button.propTypes)),
+
+  /**
+   * The descriptive text, displayed under the title.
+   */
+  message: PropTypes.string,
+
+  /**
+   * The title displayed under the glyph. Variants contain default titles that can be overridden by this prop.
+   */
+  title: PropTypes.string,
+
+  /**
+   * Sets the glyph and title using a pre-baked variant. One of the following: `no-data`,
+   * `no-matching-results`, `not-authorized`, or `error`
+   */
+  variant: PropTypes.oneOf(['no-data', 'no-matching-results', 'not-authorized', 'error']),
+};
+
+const defaultProps = {
+  buttonAttrs: [],
+  message: undefined,
+  title: undefined,
+  variant: undefined,
+};
+
+const ApplicationStatusOverlay = (props) => {
+  const {
+    buttonAttrs,
+    message,
+    title,
+    variant,
+  } = props;
+  const id = useRef(uuidv4());
```

Comment:
Nitpicking here, but one convention I've used elsewhere is the addition of a 'Ref' suffix to variables that hold refs. I find it's a good reminder to access the `current` property on it. So this would ideally be `idRef`.

I'm realizing I _didn't_ do that for the ApplicationLoadingOverlay's id variable. We should update that one too :P

Reply:
Renamed ref variables for both ApplicationStatusOverlay and ApplicationLoadingOverlay here : https://github.com/cerner/terra-application/pull/47/commits/5e92dd7f93fe585a4883c6aa4734cb2859e0fa06